<launch>
  <!-- launch the nodelet manager -->
  <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" />

  <!-- CloudLoader -->
  <!-- <node name="cloud_loader" pkg="soma_ros" type="cloud_loader" /> -->

  <!-- PassThrough -->
  <node pkg="nodelet" type="nodelet" name="passthrough_z" args="load pcl/PassThrough pcl_manager" >
    <!-- <remap from="~input" to="/cloud_raw" /> -->
    <remap from="~input" to="/camera_L/depth_registered/points"/>
    <remap from="~output" to="/cloud_passthrough_z" />
    <rosparam>
      filter_field_name: z
      filter_limit_min: 0.01
      filter_limit_max: 15.0
      filter_limit_negative: False
    </rosparam>
  </node>

  <node pkg="nodelet" type="nodelet" name="passthrough_y" args="load pcl/PassThrough pcl_manager" >
    <remap from="~input" to="/cloud_passthrough_z" />
    <remap from="~output" to="/cloud_passthrough" />
    <rosparam>
      filter_field_name: y
      filter_limit_min: 0.01
      filter_limit_max: 3.0
      filter_limit_negative: False
    </rosparam>
  </node>

  <!-- VoxelGrid -->
  <node pkg="nodelet" type="nodelet" name="voxel_grid_z" args="load pcl/VoxelGrid pcl_manager" >
    <remap from="~input" to="/cloud_filtered" />
    <remap from="~output" to="/cloud_downsampled_z" />
    <rosparam>
     filter_field_name: z
     filter_limit_min: 0.01 
     filter_limit_max: 15.0
     filter_limit_negative: False
     leaf_size: 0.01
    </rosparam>
  </node>
  
  <!-- StatisticalOutlierRemoval -->
  <node pkg="nodelet" type="nodelet" name="statistical_outlier_removal" args="load pcl/StatisticalOutlierRemoval pcl_manager" >
    <remap from="~input" to="/cloud_passthrough" />
    <remap from="~output" to="/cloud_filtered" />
    <rosparam>
      mean_k: 10
      stddev: 0.15
    </rosparam>
  </node>

  <!-- <node pkg="nodelet" type="nodelet" name="voxel_grid_y" args="load pcl/VoxelGrid pcl_manager" >
    <remap from="~input" to="/cloud_downsampled_z" />
    <remap from="~output" to="/cloud_downsampled" />
    <rosparam>
     filter_field_name: y
     filter_limit_min: 0.01 
     filter_limit_max: 3.0
     filter_limit_negative: False
     leaf_size: 0.01
    </rosparam>
  </node> -->


  <!-- PlaneSegmentation -->
  <node pkg="nodelet" type="nodelet" name="plane_segmenatation"
        args="load soma_ros/CloudPlaneSegmentator pcl_manager" >
  </node>  


  <!-- <node name="cloud_planar_segmenter" pkg="buggy_segmentation" type="cloud_planar_segmenter" /> -->
  <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find soma_ros)/rviz/tomokawa_plane_seg.rviz" required="true" /> -->
</launch>
